pre_commit_hooks/destroyed_symlinks.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import sys
+from operator import methodcaller
+from subprocess import check_call
+from subprocess import check_output
+from typing import Optional
+from typing import Sequence
+
+ORDINARY_CHANGED_ENTRIES_MARKER = b'1'
+PERMS_LINK = b'120000'
+PERMS_NONEXIST = b'000000'
+
+
+def normalize_content(content: bytes) -> bytes:
+    return b'\n'.join(
+        filter(
+            None,
+            map(
+                methodcaller('strip'),
+                content.splitlines(),
+            ),
+        ),
+    )
+
+
+def find_destroyed_symlinks(autofix: bool) -> Sequence[bytes]:
+    destroyed_links = []
+    for line in check_output(
+        ['git', 'status', '--porcelain=v2', '-z'],
+    ).split(b'\0'):
+        splitted = line.split(b' ')
```

Comment:
is this safe to space split? does `.split()` do the same?

Reply:
I guess both variants are safe, I'll change it to plain `.split()`.

Reply:
Oh, no, it's important to split by spaces because at the end there is filename which is rejoined later. If you have filenames with tabs or (more common) with more than one space in a row, you will have corrupted filename after rejoin.